src/coins.h

```diff
@@ -115,13 +115,20 @@ struct CCoinsCacheEntry
     unsigned char flags;
 
     enum Flags {
-        DIRTY = (1 << 0), // This cache entry is potentially different from the version in the parent view.
-        FRESH = (1 << 1), // The parent view does not have this entry (or it is pruned).
-        /* Note that FRESH is a performance optimization with which we can
-         * erase coins that are fully spent if we know we do not need to
-         * flush the changes to the parent cache.  It is always safe to
-         * not mark FRESH if that condition is not guaranteed.
-         */
+        //! This cache entry is potentially different from the version in the parent cache.
+        //! Failing to mark a coin as DIRTY when it is potentially different from the
+        //! parent view will cause a consensus failure, since the coin's state won't get
+        //! written to the parent when the cache is flushed.
+        //! Note that all empty coins in the cache MUST be DIRTY.
+        DIRTY = (1 << 0),
+        //! The parent cache does not have this coin (or it is spent in the parent cache).
+        //! If a FRESH coin is spent in this cache, then it can be deleted entirely
+        //! and doesn't ever need to be flushed to the parent. This is a performance
+        //! optimization. Note that a coin cannot be both FRESH and empty.
+        //! Marking a coin FRESH which is does exist in the parent view will
```

Comment:
grammar issue- "which is does exist" 
maybe- "which actually does exist" 

also, I find this sentence a bit confusing, mostly because of the phrasing of "will" vs "might". Are there other ways making a coin that exists in parent view as FRESH could cause a consensus failure?

Reply:
> grammar issue- "which is does exist"
> maybe- "which actually does exist"

Or simply "exists"?

Reply:
fixed